pragma/publisher/fetchers/starknetamm.py

```diff
@@ -176,40 +181,96 @@ def on_fetch_jedi_price_sync(self) -> float:
+                if isinstance(await self.off_fetch_ekubo_price(asset, session), float)
+                else None
+            )
+            ekubo_price = None
+            eth_usd_price = eth_usd_entry.price / (10 ** self.ETH_USD[0]["decimals"])
+            jedi_swap_price = await self.on_fetch_jedi_price(session)
+            if ekubo_price is not None and jedi_swap_price is not None:
+                price = eth_usd_price / ((ekubo_price + jedi_swap_price) / 2)
+                return self._construct(asset, price)
+            elif ekubo_price is not None:
+                price = eth_usd_price / ekubo_price
+                return self._construct(asset, price)
+            elif jedi_swap_price is not None:
+                price = eth_usd_price / jedi_swap_price
+                return self._construct(asset, price)
+            else:
+                logger.error("Both ekubo_price and jedi_swap_price are null")
+                return PublisherFetchError("Both prices are unavailable")
         else:
-            logger.error("Both ekubo_price and jedi_swap_price are null")
-            return PublisherFetchError("Both prices are unavailable")
+            logger.error("Pair not available for the Starknet fetcher")
 
     def _fetch_strk_sync(self, asset) -> SpotEntry:
-        # ekubo_price =  self.on_fetch_ekubo_price_sync()
-        ekubo_price = (
-            self.off_fetch_ekubo_price_sync(asset)
-            if isinstance(self.off_fetch_ekubo_price_sync(asset), float)
-            else None
-        )
-        jedi_swap_price = self.on_fetch_jedi_price_sync()
-        if ekubo_price is not None and jedi_swap_price is not None:
-            return self._construct(asset, (ekubo_price + jedi_swap_price) / 2)
-        elif ekubo_price is not None:
-            return self._construct(asset, ekubo_price)
-        elif jedi_swap_price is not None:
-            return self._construct(asset, jedi_swap_price)
+        if asset["pair"] == ("ETH", "STRK"):
+            # ekubo_price =  self.on_fetch_ekubo_price_sync()
+            ekubo_price = (
+                self.off_fetch_ekubo_price_sync(asset)
+                if isinstance(self.off_fetch_ekubo_price_sync(asset), float)
+                else None
+            )
+            jedi_swap_price = self.on_fetch_jedi_price_sync()
+            if ekubo_price is not None and jedi_swap_price is not None:
+                return self._construct(asset, (ekubo_price + jedi_swap_price) / 2)
+            elif ekubo_price is not None:
+                return self._construct(asset, ekubo_price)
+            elif jedi_swap_price is not None:
+                return self._construct(asset, jedi_swap_price)
+            else:
+                logger.error("Both ekubo_price and jedi_swap_price are null")
+                return PublisherFetchError("Both prices are unavailable")
+        elif asset["pair"] == ("STRK", "USD"):
+            defillama_fetcher = DefillamaFetcher(self.ETH_USD, self.publisher)
```

Comment:
I would prefer if we query it on-chain using `client.get_data` rather than adding a dependency on another fetcher, wdyt ?